pipeline/console/tags.py

```diff
@@ -0,0 +1,153 @@
+import argparse
+import re
+import sys
+from typing import List
+
+from tabulate import tabulate
+
+from pipeline import PipelineCloud
+from pipeline.schemas.pagination import Paginated
+from pipeline.schemas.pipeline import (
+    PipelineTagCreate,
+    PipelineTagGet,
+    PipelineTagPatch,
+)
+from pipeline.util.logging import _print
+
+tag_re_pattern = re.compile(r"^[a-zA-Z\-\_]+:[a-zA-Z\-\_]+$", re.IGNORECASE)
+
+
+def _get_tag(tag_name: str) -> PipelineTagGet:
+    remote_service = PipelineCloud(verbose=False)
+    remote_service.authenticate()
+    tag_information = PipelineTagGet.parse_obj(
+        remote_service._get(f"/v2/pipeline-tags/by-name/{tag_name}")
+    )
+    return tag_information
+
+
+def _update_or_create_tag(source: str, target: str, sub_command: str) -> PipelineTagGet:
+    remote_service = PipelineCloud(verbose=False)
+    remote_service.authenticate()
+    if not tag_re_pattern.match(source):
+        _print("Source tag must match pattern 'pipeline:tag'", level="ERROR")
```

Comment:
If we want to follow the `docker tag` CLI then `source` should be the 'origin' pipeline, i.e. the pipeline that the new tag `target` will point to.

```
$ docker tag create
"docker tag" requires exactly 2 arguments.
See 'docker tag --help'.

Usage:  docker tag SOURCE_IMAGE[:TAG] TARGET_IMAGE[:TAG]

Create a tag TARGET_IMAGE that refers to SOURCE_IMAGE
```

This `if not tag_re_pattern.match(source):` prevents the source (in the `docker tag` sense) being a pipeline ID.

Reply:
done